charts/sysdig/templates/_helpers.tpl

```diff
@@ -231,3 +231,20 @@ Sysdig Eve Connector Secret generation (if not exists)
 {{ randAlphaNum 32 | b64enc | quote }}
 {{- end -}}
 {{- end -}}
+
+{{/*
+to help the maxUnavailable and max_parallel_cold_starts pick a reasonable value based on the cluster size
+*/}}
+{{- define "sysdig.parallelStarts" -}}
+{{- if .Values.daemonset.updateStrategy.rollingUpdate.maxUnavailable -}}
+    {{- .Values.daemonset.updateStrategy.rollingUpdate.maxUnavailable -}}
+{{- else if eq .Values.resourceProfile "small" -}}
+    {{- 1 -}}
+{{- else if eq .Values.resourceProfile "medium" -}}
+    {{- 10 -}}
+{{- else if eq .Values.resourceProfile "large" -}}
+    {{- 20 -}}
```

Comment:
As we discussed offline, setting the `max_parallel_cold_starts` to 20 may add too much load on the Apiserver for big clusters.
Cold-start is implemented using k8s leader election mechanism based on leases resources. Each agent tries to acquire each lease every 15s during the bootstrap, this means that in a large cluster, say 1K nodes we will have ~1333rps in the worst case (1000 nodes * 20 leases / 15s). I would keep 10 for the moment that is more conservative, until we better assess the impact.

Reply:
thank you. I will change this value to 10 for all cluster with more than 10 nodes.

Reply:
@irozzo-1A did you suggest if someone from QA to test this?